addons/grenades/functions/fnc_flashbangExplosionEH.sqf

```diff
@@ -48,30 +50,40 @@ _affected = _grenade nearEntities ["CAManBase", 50];
             //Do effects for player
             // is there line of sight to the grenade?
             _posGrenade = getPosASL _grenade;
+            _eyePos = eyePos ACE_player; //PositionASL
             _posGrenade set [2, (_posGrenade select 2) + 0.2]; // compensate for grenade glitching into ground
-            if (lineIntersects [_posGrenade, getPosASL _x, _grenade, _x]) then {
+
+            //Check for line of sight:
+            if (lineIntersects [_posGrenade, _eyePos, _grenade, _x]) then {
```

Comment:
This may greaty reduce the effect if small props are near the grenade. Maybe we could potentially create 4~5 origin positions by applying 0.25 m offsets in various directions and check all of those for visibility. Effect would still be full unless all of them have intersections.